vm/devices/storage/disk_crypt/block_crypto/src/lib.rs

```diff
@@ -0,0 +1,328 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Cryptography primitives for disk encryption.
+
+#![warn(missing_docs)]
+
+#[cfg(windows)]
+use bcrypt as sys;
+#[cfg(unix)]
+use ossl as sys;
+use thiserror::Error;
+
+/// AES-XTS-256 encryption/decryption.
+pub struct AesXts256(sys::AesXts256);
+
+/// An error for cryptographic operations.
+#[derive(Debug, Error)]
+#[error(transparent)]
+pub struct Error(#[from] ErrorInner);
+
+#[derive(Debug, Error)]
+enum ErrorInner {
+    #[error("invalid key length")]
+    InvalidKeyLength,
+    #[error("platform crypto error")]
+    Sys(sys::Error),
+}
+
+impl AesXts256 {
+    /// The required key length for the algorithm.
+    pub const KEY_LEN: usize = 64;
+
+    /// Creates a new AES-XTS-256 encryption/decryption context.
+    pub fn new(key: &[u8], data_unit_size: u32) -> Result<Self, Error> {
```

Comment:
this ties into other discussions we had, but I would prefer having this API take a `&[u8; KEY_LEN]`, and having the caller try-into if need be

Reply:
Yeah fair enough.